homeassistant/components/mqtt/models.py

```diff
@@ -289,12 +295,51 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    __slots__ = ("_entity", "_attributes")
+
+    def __init__(self, entity: Entity) -> None:
+        """Initialize entity monitor."""
+        self._entity: Entity = entity
+        self._attributes: dict[str, Any] = {}
+
+    def track(self, attributes: set[str]) -> None:
+        """Start tracking attributes."""
+        self._attributes = {
+            attribute: getattr(self._entity, attribute, UNDEFINED)
+            for attribute in attributes
+        }
+
+    @property
+    def assume_has_changed(self) -> bool:
+        """Return True if attributes on entity changed or if update is forced.
+
+        Stops tracking attribute changes.
+        """
+        entity = self._entity
+        attributes = self._attributes
+        if not (
+            assume_has_changed := (
+                getattr(entity, "_attr_force_update", False) or not attributes
+            )
+        ):
```

Comment:
If no monitoring attributes are set, we will call `async_ha_write_state` and let the core decide.
If `_attr_force_update` is set, the state should always be updated regardless of changes.

Reply:
But if we assign the new value, it should not be an issue.